### PR TITLE
Add PoffinCooker

### PR DESCRIPTION
**Pull request by mb-baboul**
Add PoffinCooker for BDSP poffin cooking mini-game.

Also add a new setting for CurryHunter program to improve the detection of a curry Pokémon.

### CODE REVIEW

File: SerialPrograms/Source/PokemonBDSP/Programs/Farming/PokemonBDSP_PoffinCooker.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    return std::unique_ptr<StatsTracker>(new Stats());
+}
+
+
+PoffinCooker::PoffinCooker()
+    : GO_HOME_WHEN_DONE(false)
+    , MAX_COOK_ATTEMPTS(
+        "<b>Cook this many times:</b><br>This puts a limit on how many poffins you get. Don't forget that each cooking session gets you 4 poffins, and your bag cannot have more than 100 poffins. Thus you should never input more than 25 here.",
+        LockWhileRunning::LOCKED,
+        5
```
Comment (pifopi) [inline SerialPrograms/Source/PokemonBDSP/Programs/Farming/PokemonBDSP_PoffinCooker.cpp:52]: You can have 25 as max in code to make sure users don't try dumb things

File: SerialPrograms/Source/PokemonBDSP/Programs/Farming/PokemonBDSP_PoffinCooker.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    //  Select the cooking option.
+    pbf_press_button(context, BUTTON_A, 5, 100);
+    pbf_press_button(context, BUTTON_A, 5, 175);
+    pbf_press_button(context, BUTTON_A, 5, 75);
+    pbf_press_button(context, BUTTON_A, 5, 100);
+
+
+    for (uint16_t c = 0; c < MAX_COOK_ATTEMPTS; c++) {
+
+        env.log("Select the 4 berrie to use");
```
Comment (pifopi) [inline SerialPrograms/Source/PokemonBDSP/Programs/Farming/PokemonBDSP_PoffinCooker.cpp:131]: berries*

**Review by pifopi — APPROVED** [review]

File: SerialPrograms/Source/PokemonBDSP/Programs/Farming/PokemonBDSP_PoffinCooker.cpp (hunk 1 above)
Comment (mb-baboul) [reply SerialPrograms/Source/PokemonBDSP/Programs/Farming/PokemonBDSP_PoffinCooker.cpp]: Added to the code

File: SerialPrograms/Source/PokemonBDSP/Programs/Farming/PokemonBDSP_PoffinCooker.cpp (hunk 2 above)
Comment (mb-baboul) [reply SerialPrograms/Source/PokemonBDSP/Programs/Farming/PokemonBDSP_PoffinCooker.cpp]: Fixed

**Pull request merged**